In [1]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
%matplotlib inline


In [2]:
# @hidden_cell
# The project token is an authorization token that is used to access project resources like data sources, connections, and used by platform APIs.
from project_lib import Project
project = Project(project_id='70028ea9-2749-4f47-81e4-75651292d998', project_access_token='p-ec4a80cb9bd43b283ee0a424ea141bed63d3f8c1')
pc = project.project_context


In [3]:
my_file = project.get_file("Motor_Vehicle_Collisions.csv")
my_file.seek(0)
df = pd.read_csv(my_file)
df.head()


/opt/conda/envs/Python36/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3020: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


DATE  TIME   BOROUGH ZIP CODE   LATITUDE  LONGITUDE  \
0  7/12/2019  0:00     BRONX    10472  40.831623  -73.86726   
1  7/12/2019  0:00  BROOKLYN    11207  40.669792  -73.89240   
2  7/12/2019  0:00  BROOKLYN    11234  40.618717  -73.93262   
3  7/12/2019  0:00       NaN      NaN  40.829155  -73.93728   
4  7/12/2019  0:00       NaN      NaN  40.718987  -73.94635   

                 LOCATION                    ON STREET NAME  \
0  (40.831623, -73.86726)  WESTCHESTER AVENUE                 
1   (40.669792, -73.8924)  SUTTER AVENUE                      
2  (40.618717, -73.93262)                               NaN   
3  (40.829155, -73.93728)  8 AVENUE                           
4  (40.718987, -73.94635)  BROOKLYN QUEENS EXPRESSWAY         

       CROSS STREET NAME                           OFF STREET NAME  ...  \
0  SAINT LAWRENCE AVENUE                                       NaN  ...   
1           WYONA STREET                                       NaN  ...   
2                    NaN  2098      FLATBUSH AVENUE                 ...   
3                    NaN                                       NaN  ...   
4                    NaN                                       NaN  ...   

   CONTRIBUTING FACTOR VEHICLE 2  CONTRIBUTING FACTOR VEHICLE 3  \
0                            NaN                            NaN   
1                    Unspecified                            NaN   
2                    Unspecified                            NaN   
3                    Unspecified                            NaN   
4                    Unspecified                            NaN   

   CONTRIBUTING FACTOR VEHICLE 4  CONTRIBUTING FACTOR VEHICLE 5  UNIQUE KEY  \
0                            NaN                            NaN     4168556   
1                            NaN                            NaN     4169745   
2                            NaN                            NaN     4169938   
3                            NaN                            NaN     4169906   
4                            NaN                            NaN     4168595   

                   VEHICLE TYPE CODE 1                  VEHICLE TYPE CODE 2  \
0                                Sedan                                  NaN   
1  Station Wagon/Sport Utility Vehicle                                 Bike   
2  Station Wagon/Sport Utility Vehicle  Station Wagon/Sport Utility Vehicle   
3                                Sedan  Station Wagon/Sport Utility Vehicle   
4                                Sedan                                Sedan   

   VEHICLE TYPE CODE 3 VEHICLE TYPE CODE 4 VEHICLE TYPE CODE 5  
0                  NaN                 NaN                 NaN  
1                  NaN                 NaN                 NaN  
2                  NaN                 NaN                 NaN  
3                  NaN                 NaN                 NaN  
4                  NaN                 NaN                 NaN  

[5 rows x 29 columns]

In [4]:
df.dtypes

DATE                              object
TIME                              object
BOROUGH                           object
ZIP CODE                          object
LATITUDE                         float64
LONGITUDE                        float64
LOCATION                          object
ON STREET NAME                    object
CROSS STREET NAME                 object
OFF STREET NAME                   object
NUMBER OF PERSONS INJURED        float64
NUMBER OF PERSONS KILLED         float64
NUMBER OF PEDESTRIANS INJURED      int64
NUMBER OF PEDESTRIANS KILLED       int64
NUMBER OF CYCLIST INJURED          int64
NUMBER OF CYCLIST KILLED           int64
NUMBER OF MOTORIST INJURED         int64
NUMBER OF MOTORIST KILLED          int64
CONTRIBUTING FACTOR VEHICLE 1     object
CONTRIBUTING FACTOR VEHICLE 2     object
CONTRIBUTING FACTOR VEHICLE 3     object
CONTRIBUTING FACTOR VEHICLE 4     object
CONTRIBUTING FACTOR VEHICLE 5     object
UNIQUE KEY                         int64
VEHICLE TYPE COD

In [5]:
# size of dataframe (rows, columns)
df.shape    

(1048575, 29)

In [6]:
df_col= df[['BOROUGH', 'LATITUDE', 'LONGITUDE', 'DATE', 'NUMBER OF PERSONS INJURED', 'NUMBER OF PERSONS KILLED', 'NUMBER OF PEDESTRIANS INJURED', 'NUMBER OF PEDESTRIANS KILLED', 'NUMBER OF CYCLIST INJURED', 'NUMBER OF CYCLIST KILLED', 'NUMBER OF MOTORIST INJURED', 'NUMBER OF MOTORIST KILLED', 'CONTRIBUTING FACTOR VEHICLE 1']]

In [7]:
df_col.tail(5)

BOROUGH   LATITUDE  LONGITUDE        DATE  NUMBER OF PERSONS INJURED  \
1048570   BRONX  40.816775 -73.809126  11/16/2014                        1.0   
1048571  QUEENS  40.766446 -73.829089  11/16/2014                        0.0   
1048572     NaN  40.759949 -73.838690  11/16/2014                        0.0   
1048573     NaN        NaN        NaN  11/16/2014                        1.0   
1048574     NaN  40.759949 -73.838690  11/16/2014                        0.0   

         NUMBER OF PERSONS KILLED  NUMBER OF PEDESTRIANS INJURED  \
1048570                       0.0                              1   
1048571                       0.0                              0   
1048572                       0.0                              0   
1048573                       0.0                              0   
1048574                       0.0                              0   

         NUMBER OF PEDESTRIANS KILLED  NUMBER OF CYCLIST INJURED  \
1048570                             0                          0   
1048571                             0                          0   
1048572                             0                          0   
1048573                             0                          0   
1048574                             0                          0   

         NUMBER OF CYCLIST KILLED  NUMBER OF MOTORIST INJURED  \
1048570                         0                           0   
1048571                         0                           0   
1048572                         0                           0   
1048573                         0                           1   
1048574                         0                           0   

         NUMBER OF MOTORIST KILLED   CONTRIBUTING FACTOR VEHICLE 1  
1048570                          0                     Unspecified  
1048571                          0                     Unspecified  
1048572                          0  Driver Inattention/Distraction  
1048573                          0                     Unspecified  
1048574                          0                 Other Vehicular

In [8]:
# Drop all rows that have any NaN (missing) values, drop the rows even with single NaN or single missing values.
df_col = df_col.dropna()
df_col.tail(2)


BOROUGH   LATITUDE  LONGITUDE        DATE  NUMBER OF PERSONS INJURED  \
1048570   BRONX  40.816775 -73.809126  11/16/2014                        1.0   
1048571  QUEENS  40.766446 -73.829089  11/16/2014                        0.0   

         NUMBER OF PERSONS KILLED  NUMBER OF PEDESTRIANS INJURED  \
1048570                       0.0                              1   
1048571                       0.0                              0   

         NUMBER OF PEDESTRIANS KILLED  NUMBER OF CYCLIST INJURED  \
1048570                             0                          0   
1048571                             0                          0   

         NUMBER OF CYCLIST KILLED  NUMBER OF MOTORIST INJURED  \
1048570                         0                           0   
1048571                         0                           0   

         NUMBER OF MOTORIST KILLED CONTRIBUTING FACTOR VEHICLE 1  
1048570                          0                   Unspecified  
1048571                          0                   Unspecified

In [9]:
df_col.BOROUGH.value_counts()

BROOKLYN         211793
QUEENS           183346
MANHATTAN        152546
BRONX             96756
STATEN ISLAND     27072
Name: BOROUGH, dtype: int64

In [10]:
df_col.BOROUGH.value_counts(normalize = True)

BROOKLYN         0.315397
QUEENS           0.273034
MANHATTAN        0.227168
BRONX            0.144087
STATEN ISLAND    0.040315
Name: BOROUGH, dtype: float64

In [11]:
df_col.rename(columns={'NUMBER OF PERSONS INJURED': 'NUM_INJURED',
                       'NUMBER OF PERSONS KILLED': 'NUM_KILLED', 
                       'NUMBER OF PEDESTRIANS INJURED': 'NUM_PED_INJURED',
                       'NUMBER OF PEDESTRIANS KILLED': 'NUM_PED_KILLED',
                       'NUMBER OF CYCLIST INJURED': 'NUM_CYC_INJURED',
                       'NUMBER OF CYCLIST KILLED': 'NUM_CYC_KILLED',
                       'NUMBER OF MOTORIST INJURED': 'NUM_MOTORIST_INJURED',
                       'NUMBER OF MOTORIST KILLED': 'NUM_MOTORIST_KILLED',
                       'CONTRIBUTING FACTOR VEHICLE 1': 'FACTOR_V_1'}, inplace=True)
df_col.columns

Index(['BOROUGH', 'LATITUDE', 'LONGITUDE', 'DATE', 'NUM_INJURED', 'NUM_KILLED',
       'NUM_PED_INJURED', 'NUM_PED_KILLED', 'NUM_CYC_INJURED',
       'NUM_CYC_KILLED', 'NUM_MOTORIST_INJURED', 'NUM_MOTORIST_KILLED',
       'FACTOR_V_1'],
      dtype='object')

In [12]:
pd.crosstab(df_col.BOROUGH, df_col.FACTOR_V_1)

FACTOR_V_1     1  80  Accelerator Defective  Aggressive Driving/Road Rage  \
BOROUGH                                                                     
BRONX          0   6                     67                           536   
BROOKLYN       0  16                    123                           878   
MANHATTAN      1   3                     52                           606   
QUEENS         1  20                    101                           642   
STATEN ISLAND  1   1                     24                           110   

FACTOR_V_1     Alcohol Involvement  Animals Action  Backing Unsafely  \
BOROUGH                                                                
BRONX                         1111              59              5016   
BROOKLYN                      2233             115             10270   
MANHATTAN                      987              39              6287   
QUEENS                        2153             141             11159   
STATEN ISLAND                  416             120              1639   

FACTOR_V_1     Brakes Defective  Cell Phone (hand-Held)  \
BOROUGH                                                   
BRONX                       388                      42   
BROOKLYN                    704                      49   
MANHATTAN                   318                      27   
QUEENS                      525                      36   
STATEN ISLAND               154                      10   

FACTOR_V_1     Cell Phone (hand-held)  ...  \
BOROUGH                                ...   
BRONX                               2  ...   
BROOKLYN                           10  ...   
MANHATTAN                           9  ...   
QUEENS                              6  ...   
STATEN ISLAND                       1  ...   

FACTOR_V_1     Traffic Control Device Improper/Non-Working  \
BOROUGH                                                      
BRONX                                                   38   
BROOKLYN                                                84   
MANHATTAN                                               51   
QUEENS                                                  70   
STATEN ISLAND                                           14   

FACTOR_V_1     Traffic Control Disregarded  Turning Improperly  \
BOROUGH                                                          
BRONX                                 1647                2600   
BROOKLYN                              3788                4490   
MANHATTAN                             1628                5737   
QUEENS                                3945                4165   
STATEN ISLAND                          518                 705   

FACTOR_V_1     Unsafe Lane Changing  Unsafe Speed  Unspecified  \
BOROUGH                                                          
BRONX                          1753          1114        37218   
BROOKLYN                       2775          2045        77666   
MANHATTAN                      5060           716        38612   
QUEENS                         2124          1940        54774   
STATEN ISLAND                   347           382         7176   

FACTOR_V_1     Using On Board Navigation Device  Vehicle Vandalism  \
BOROUGH                                                              
BRONX                                        10                 11   
BROOKLYN                                     12                 23   
MANHATTAN                                     5                  6   
QUEENS                                        8                  9   
STATEN ISLAND                                 2                  1   

FACTOR_V_1     View Obstructed/Limited  Windshield Inadequate  
BOROUGH                                                        
BRONX                              882                      4  
BROOKLYN                          1612                     11  
MANHATTAN                          850                      4  
QUEENS                            1374        

In [15]:
# We can check to see how many null objects we have in the dataset as follows:
df_col.isnull().sum()

BOROUGH                 0
LATITUDE                0
LONGITUDE               0
DATE                    0
NUM_INJURED             0
NUM_KILLED              0
NUM_PED_INJURED         0
NUM_PED_KILLED          0
NUM_CYC_INJURED         0
NUM_CYC_KILLED          0
NUM_MOTORIST_INJURED    0
NUM_MOTORIST_KILLED     0
FACTOR_V_1              0
dtype: int64

In [16]:
# Finally, let's view a quick summary of each column in our dataframe using the `describe()` method.
df_col.describe()

LATITUDE      LONGITUDE    NUM_INJURED     NUM_KILLED  \
count  671513.000000  671513.000000  671513.000000  671513.000000   
mean       40.676022     -73.835956       0.244144       0.000929   
std         1.379516       2.501442       0.625736       0.032595   
min         0.000000     -74.253930       0.000000       0.000000   
25%        40.667522     -73.976759       0.000000       0.000000   
50%        40.719368     -73.929886       0.000000       0.000000   
75%        40.765676     -73.869210       0.000000       0.000000   
max        41.126150       0.000000      32.000000       8.000000   

       NUM_PED_INJURED  NUM_PED_KILLED  NUM_CYC_INJURED  NUM_CYC_KILLED  \
count    671513.000000   671513.000000    671513.000000   671513.000000   
mean          0.054644        0.000550         0.023285        0.000088   
std           0.240607        0.024673         0.152006        0.009531   
min           0.000000        0.000000         0.000000        0.000000   
25%           0.000000        0.000000         0.000000        0.000000   
50%           0.000000        0.000000         0.000000        0.000000   
75%           0.000000        0.000000         0.000000        0.000000   
max          27.000000        6.000000         3.000000        2.000000   

       NUM_MOTORIST_INJURED  NUM_MOTORIST_KILLED  
count          671513.00000        671513.000000  
mean                0.16629             0.000293  
std                 0.57889             0.017808  
min                 0.00000             0.000000  
25%                 0.00000             0.000000  
50%                 0.00000             0.000000  
75%                 0.00000             0.000000  
max                30.00000             3.000000

Before we proceed, notice that the defaul index of the dataset is a numeric range from 0 to 14. This makes it very difficult to do a query by a specific Borough.

This can be fixed very easily by setting the 'Borough' column as the index using `set_index()` method.

In [28]:
df_col.set_index('BOROUGH', inplace=True)
# tip: The opposite of set is reset. So to reset the index, we can use df_can.reset_index()

In [29]:
df_col.head(3)

LATITUDE  LONGITUDE       DATE  NUM_INJURED  NUM_KILLED  \
BOROUGH                                                              
BRONX     40.831623  -73.86726  7/12/2019          0.0         0.0   
BROOKLYN  40.669792  -73.89240  7/12/2019          0.0         0.0   
BROOKLYN  40.618717  -73.93262  7/12/2019          0.0         0.0   

          NUM_PED_INJURED  NUM_PED_KILLED  NUM_CYC_INJURED  NUM_CYC_KILLED  \
BOROUGH                                                                      
BRONX                   0               0                0               0   
BROOKLYN                0               0                0               0   
BROOKLYN                0               0                0               0   

          NUM_MOTORIST_INJURED  NUM_MOTORIST_KILLED            FACTOR_V_1  
BOROUGH                                                                    
BRONX                        0                    0           Unspecified  
BROOKLYN                     0                    0   Passing Too Closely  
BROOKLYN                     0                    0  Unsafe Lane Changing

In [ ]:
# optional: to remove the name of the index
df_col.index.name = None

In [30]:
# Lets view the number of deaths and injuries from BROOKLYN :
# 1. the full row data (all columns)
print(df_col.loc['BROOKLYN'])

# alternate methods
print(df_col[df_col.index == 'BROOKLYN'].T.squeeze())

           LATITUDE  LONGITUDE        DATE  NUM_INJURED  NUM_KILLED  \
BOROUGH                                                               
BROOKLYN  40.669792 -73.892400   7/12/2019          0.0         0.0   
BROOKLYN  40.618717 -73.932620   7/12/2019          0.0         0.0   
BROOKLYN  40.666733 -73.950790   7/12/2019          0.0         0.0   
BROOKLYN  40.613796 -73.979750   7/12/2019          0.0         0.0   
BROOKLYN  40.624474 -74.006680   7/12/2019          0.0         0.0   
BROOKLYN  40.671574 -73.916520   7/12/2019          0.0         0.0   
BROOKLYN  40.637985 -73.899680   7/12/2019          0.0         0.0   
BROOKLYN  40.641730 -73.963264   7/12/2019          1.0         0.0   
BROOKLYN  40.648370 -73.945420   7/12/2019          0.0         0.0   
BROOKLYN  40.649487 -73.887210   7/12/2019          0.0         0.0   
BROOKLYN  40.650387 -73.958720   7/12/2019          0.0         0.0   
BROOKLYN  40.623610 -73.931280   7/12/2019          0.0         0.0   
BROOKL

# 1. Introduction

An insurance company is trying to launch an office in New York city in one of the Borough's to make profit at a minimum cost. There are five Borough categories; Bronx, Brooklyn, Manhattan, Queens and Staten Island. The company wants to decide one Borough out of these five locations. For the insurance company's profitability and set insurance premium for the residents of New York city, and advocate a systems approach to road traffic safety that emphasizes involvement at all levels of the road traffic system, studying motor vehicle collisions statistics from public dataset in each Borough group is important. The dataset contains information on motorists, cyclists, and pedestrians. The company is trying to answer which regions are better or worse in terms of number of injuries and fatalities?  
   
Road traffic accidents are the leading cause of death and injury and the tenth leading cause of all deaths globally. Now make up a surprisingly significant portion of the worldwide burden of ill health. An estimated 1.2 million people are killed in road crashes each year and as many as 50 million are injured. If present trends continue, road traffic injuries are predicted to be the third leading contributor to the global burden of disease and injury by 2020. This burden is creating enormous economic hardship due to the loss of family.  
Road traffic injuries are preventable if successsful interventions such as seat belt safety laws, enforcement of speed limits, warnings about the dangers of mixing alcohol consumption with driving, and safer design and use of roads and vehicles are considered. More than one-half of all road traffic deaths globally occur among people ages 15 to 44—their most productive earning years. Moreover, the disability burden for this age group accounts for 60 percent of all lost because of road traffic accidents. 
Driver impairment is an important component of road traffic accidents, driving at excess speeds, while under the influence of alcohol or drugs, while sleepy or tired, when visibility is compromised, or without protective gear for all vehicle occupants are major factors in crashes, deaths, and serious injuries.
Prevention interventions include: managing risk exposure to potential road traffic injury, Planning and designing roads for safety, setting road and safety rules, securing compliance, and improving transport policy, Providing visible, crashworthy, and smart vehicles.

The dataset contains a breakdown of every collision by location and injury. Each record represents a collision in New York by borough and this data can be used to do in-depth analysis.
This would interest any insurance company to open an office in New york city.



# 2. Dataset

In this section the dataset used in the analysis is described. The public dataset about motor vehicle collision was obtained from kaggle website and there are 29 variables
and 1.4 million rows. Each record includes location coordinates and other feauture variables, most notably the number of injuries and fatalities, segmented further by motorists, cyclists, and pedestrians. Out of these feature variables some of them important for this particular study are selected. These are Borough, Latitude, Longitude, number of persons injured, number of persons killed, number of pedestrians injured, number of pedestrians killed, number of cyclist injured, number of cyclist killed, number of motorist injured and number of motorist killed. The original dataset is processed, the rows that has any missing values were removed in order to perform statistical analysis. The variable Borough is used to study which of the Borough group has the higher number of deaths and injuries as comparred to the other. The motor vehicle collisions dataset is in .csv format, which can be easily imported into pandas. After the dataset is processed and the missing values were removed, the pandas dataframe contains information on 671513 records; from these 211793 Brooklyn, 183346 Queens, 152546 Manhattan, 96756 Bronx and 27072 Staten Island. 

For a Borough that has the lowest number of people killed and injured, a Foursquare location data is used in order to get the most common venues of given Borough of New York city and then we will use these features to cluster the Borough's in to different cluster groups.   

The total number of people involved in accident by each Borough category, the maximum number of people killed as well as the average number of person's killed is presented in the following output:


In [25]:
df_col.groupby('BOROUGH').NUM_KILLED.agg(['count', 'min', 'max', 'mean', 'median'])       

count  min  max      mean  median
BOROUGH                                          
BRONX           96756  0.0  2.0  0.000837     0.0
BROOKLYN       211793  0.0  2.0  0.000977     0.0
MANHATTAN      152546  0.0  8.0  0.000741     0.0
QUEENS         183346  0.0  4.0  0.001004     0.0
STATEN ISLAND   27072  0.0  2.0  0.001441     0.0

Furthermore, the following table shows the total number of people involved in accident by each Borough category, the maximum number of people injured as well as the average number of person's injured. The highest number of people injured is in the District(Borough) of Queens which is 32 where as the lowest is 20 in Bronx.

In [1]:
df_col.groupby('BOROUGH').NUM_INJURED.agg(['count', 'min', 'max', 'mean', 'median'])

NameError: name 'df_col' is not defined

In [ ]:
# After dropping unwanted columns and cleaning up NaNs, the number of deaths can be plotted by Borough:
ax = df_col[['BOROUGH','NUM_INJURED']].plot(kind='bar', title ="Number of injury", figsize=(15, 10), legend=True, fontsize=12)
ax.set_xlabel("borough", fontsize=12)
ax.set_ylabel("number injured", fontsize=12)
plt.show()

In [ ]:
df_col.plot(kind='bar',x='BOROUGH',y='NUM_INJURED')
